framework/src/userobject/ElementQualityChecker.C

```diff
@@ -0,0 +1,201 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "ElementQualityChecker.h"
+#include "MooseError.h"
+
+#include "libmesh/elem_quality.h"
+#include "libmesh/enum_elem_quality.h"
+
+MooseEnum
+ElementQualityChecker::QualityMetricType()
+{
+  return MooseEnum("ASPECT_RATIO SKEW SHEAR SHAPE MAX_ANGLE MIN_ANGLE CONDITION DISTORTION TAPER "
+                   "WARP STRETCH DIAGONAL ASPECT_RATIO_BETA ASPECT_RATIO_GAMMA SIZE JACOBIAN");
+}
+
+MooseEnum
+ElementQualityChecker::FailureMessageType()
+{
+  return MooseEnum("WARNING ERROR", "WARNING");
+}
+
+template <>
+InputParameters
+validParams<ElementQualityChecker>()
+{
+  InputParameters params = validParams<ElementUserObject>();
+  params.addClassDescription("Class to check the quality of each element using different metrics "
+                             "from libmesh.");
+
+  params.addRequiredParam<MooseEnum>("metric_type",
+                                     ElementQualityChecker::QualityMetricType(),
+                                     "Type of quality metric to be checked");
+  params.addParam<Real>("upper_bound", "the upper bound for provided metric type");
+  params.addParam<Real>("lower_bound", "The lower bound for provided metric type");
+  params.addParam<MooseEnum>("failure_type",
+                             ElementQualityChecker::FailureMessageType(),
+                             "The way how the failure of quality metric check should respond");
+  params.set<MultiMooseEnum>("execute_on") = "initial";
+
+  return params;
+}
+
+ElementQualityChecker::ElementQualityChecker(const InputParameters & parameters)
+  : ElementUserObject(parameters),
+    _m_type(getParam<MooseEnum>("metric_type").getEnum<ElemQuality>()),
+    _has_upper_bound(isParamValid("upper_bound")),
+    _has_lower_bound(isParamValid("lower_bound")),
+    _upper_bound(_has_upper_bound ? getParam<Real>("upper_bound") : 0.0),
+    _lower_bound(_has_lower_bound ? getParam<Real>("lower_bound") : 0.0),
+    _failure_type(getParam<MooseEnum>("failure_type").getEnum<FailureType>())
+{
+  if (_has_lower_bound && _has_upper_bound && _lower_bound >= _upper_bound)
```

Comment:
This check isn't good enough. If you only supply a `_lower_bound` and it's positive, you won't hit this for instance.